### PR TITLE
[PLAT-11999] Verbose logging

### PR DESCRIPTION
**Pull request by joshedney**
## Goal

Update the logging for the Bugsnag CLI to support additional log levels.

Integrate a new log wrapper that will allow us to drop different logging libraries in without much reconfiguration throughout the CLI.

Move over to using `logrus` for logging

When the `--verbose` and `--dry-run` flag are passed, the CLI prints out the payload to be sent to the API, minus the file data.

e.g.

```
[INFO] (dryrun) Skipping upload of libunity.so to https://upload.bugsnag.com:443/ndk-symbol
[INFO] (dryrun) Upload payload:
{
    "apiKey": "APIKEY",
    "appId": "com.DefaultCompany.Myproject3",
    "sharedObjectName": "libunity.so",
    "versionCode": "1",
    "versionName": "0.1"
}
```
Move information messages down to the debug log level so that they only print when passing the `--verbose` flag.

## Testing

Covered by CI

### CODE REVIEW

File: pkg/log/logrus-wrapper.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+func (l *LogrusLogger) addContextCommonFields(fields map[string]interface{}) {
+	if l.ctx != nil {
+		for k, v := range l.ctx.Value("commonFields").(map[string]interface{}) {
+			if _, ok := fields[k]; !ok {
+				fields[k] = v
+			}
+		}
+	}
+}
```
Comment (tomlongridge) [inline pkg/log/logrus-wrapper.go:98]: Is this used?

File: pkg/server/request.go [hunk 2]
```diff
@@ -83,28 +83,31 @@ func buildFileRequest(url string, fieldData map[string]string, fileFieldData map
...
 		if err != nil {
 			if strings.Contains(err.Error(), "409") {
-				log.Warn(fmt.Sprintf("Duplicate file detected, skipping upload of %s", filepath.Base(fileName)))
+				logger.Warn(fmt.Sprintf("Duplicate file detected, skipping upload of %s", filepath.Base(fileName)))
 			} else {
 				return err
 			}
 		} else {
-			log.Success(fmt.Sprintf("Uploaded %s", filepath.Base(fileName)))
+			logger.Info("Uploaded " + filepath.Base(fileName))
```
Comment (tomlongridge) [inline pkg/server/request.go:104]: Deliberate removal of sprintf?

File: pkg/upload/all.go [hunk 3]
```diff
@@ -20,18 +20,19 @@ func All(
...
 
 	fileList, err := utils.BuildFileList(paths)
 
 	if err != nil {
-		log.Error(" error building file list", 1)
+		logger.Fatal(" error building file list")
 	}
 
-	log.Info("File list built..")
+	logger.Info("File list built..")
```
Comment (tomlongridge) [inline pkg/upload/all.go:35]: Feels like the two `Info` log calls are unnecessary, when comparing this to other commands. Just an error message if it fails seems sufficient.

Likewise with the Dart command.

File: pkg/upload/all.go [hunk 4]
```diff
@@ -20,18 +20,19 @@ func All(
...
 
 	// Build the file list from the path(s)
-	log.Info("building file list...")
+	logger.Info("building file list...")
 
 	fileList, err := utils.BuildFileList(paths)
 
 	if err != nil {
-		log.Error(" error building file list", 1)
+		logger.Fatal(" error building file list")
```
Comment (tomlongridge) [inline pkg/upload/all.go:32]: We should also include the error message

```suggestion
		logger.Fatal("Error building file list")
```

File: pkg/upload/dart.go [hunk 5]
```diff
@@ -125,7 +126,7 @@ func Dart(
 
 			continue
 		}
-		log.Info(fmt.Sprintf("Skipping %s", file))
+		logger.Debug(fmt.Sprintf("Skipping %s", file))
```
Comment (tomlongridge) [inline pkg/upload/dart.go:129]:
```suggestion
		logger.Debug(fmt.Sprintf("Skipping %s - unsupported platform", file))
```

File: pkg/upload/dsym.go [hunk 6]
```diff
@@ -146,14 +147,14 @@ func ProcessDsym(
...
-					log.Info(fmt.Sprintf("Using API key from Info.plist: %s", apiKey))
+					logger.Debug(fmt.Sprintf("Using API key from Info.plist: %s", apiKey))
 				}
 			}
 		}
 
 		for _, dsym := range dwarfInfo {
 			dsymInfo := fmt.Sprintf("(UUID: %s, Name: %s, Arch: %s)", dsym.UUID, dsym.Name, dsym.Arch)
-			log.Info(fmt.Sprintf("Uploading dSYM %s", dsymInfo))
+			logger.Info(fmt.Sprintf("Uploading dSYM %s", dsymInfo))
```
Comment (tomlongridge) [inline pkg/upload/dsym.go:157]: Given the "Uploading ..." message is emitted from `ProcessFileRequest`, perhaps make this debug and the message a bit different?

```suggestion
			logger.Debug(fmt.Sprintf("Processing dSYM %s", dsymInfo))
```

File: pkg/server/request.go [hunk 7]
```diff
@@ -83,28 +83,31 @@ func buildFileRequest(url string, fieldData map[string]string, fileFieldData map
...
 		}
 	} else {
-		log.Info(fmt.Sprintf("(dryrun) Skipping upload of %s to %s", filepath.Base(fileName), endpoint))
+		logger.Info(fmt.Sprintf("(dryrun) Skipping upload of %s to %s", filepath.Base(fileName), endpoint))
+		logger.Info("(dryrun) Upload payload:")
+		prettyUploadOptions, _ := utils.PrettyPrintMap(uploadOptions)
```
Comment (tomlongridge) [inline pkg/server/request.go:109]: > When the --verbose and --dry-run flag are passed, the CLI prints out the payload to be sent to the API, minus the file data.

Does this correctly omit the file data as described?

File: pkg/server/request.go (hunk 2 above)
Comment (joshedney) [reply pkg/server/request.go]: Restored sprintf use

File: pkg/server/request.go (hunk 7 above)
Comment (joshedney) [reply pkg/server/request.go:109]: Yes, `uploadOptions` doesn't contain the file data, but does include the name of the file that is being uploaded. The file data is stored within `fileFieldData`

**Review by tomlongridge — APPROVED** [review]

**Pull request merged**